client.go

```diff
@@ -154,6 +142,14 @@ func (c *Client) ObtainCertificateUsingCSR(ctx context.Context, account acme.Acc
 		c.Logger.Info("validations succeeded; finalizing order", zap.String("order", order.Location))
 	}
 
+	csr, err := finalizer.Finalize(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("finalizing CSR: %w", err)
+	}
+
+	// TODO(hs): add validation of CSR by comparing against provided
+	// identifiers?
```

Comment:
Might as well do this before merging, if we end up keeping current API.

Reply:
Added this in d26f7c5220ef5dd8b7b9993830709fc7bd8d3005. 

For the existing `ObtainCertificateUsingCSR`, the identifiers will now be extracted from the CSR twice: once to get the right values, and the second time to validate them. This will never fail for this method; just adds a bit of time to the method execution.

There's currently no way to skip the validation. It could be an option to add (later), but I think it's OK to be strict by default and for now.

Reply:
I think that's reasonable. In the general case, the validation will be necessary. Thanks!